File: packages/contracts/python/noa_contracts/inspiration_board.py
from __future__ import annotations

from typing import Literal
from pydantic import BaseModel


class InspirationBoard(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    project_id: str
    name: str
    description: str | None
    item_count: int
    is_primary: bool
    style_profile_id: str | None
    style_inference_status: Literal[
        "pending", "running", "ready", "stale", "failed"
    ]
    created_at: str
    updated_at: str

File: packages/contracts/v1/inspiration-board.ts
import type { UUID, ISO8601 } from "./common.js";

export interface InspirationBoard {
  schema_version: "1.0";
  id: UUID;
  project_id: UUID;
  name: string;
  description: string | null;
  item_count: number;
  is_primary: boolean;
  style_profile_id: UUID | null;
  style_inference_status:
    | "pending"
    | "running"
    | "ready"
    | "stale"
    | "failed";
  created_at: ISO8601;
  updated_at: ISO8601;
}
